### PR TITLE
Finish two related files: w_012 and w_034

### PR DESCRIPTION
**Pull request by joshlory**
* https://decomp.me/scratch/h62LS
* https://decomp.me/scratch/FhCre
* https://decomp.me/scratch/N3tj5

### CODE REVIEW

File: src/weapon/w_034.c [hunk 1]
```diff
@@ -1,15 +1,97 @@
...
-extern s32 g_HandId;
-#include "shared.h"
 #include "w_034_1.h"
 #include "w_034_2.h"
 #define g_Animset w_034_1
 #define g_Animset2 w_034_2
 
-extern WeaponAnimation D_F2000_8017A934[];
+extern u8 D_8007341E;
+extern u8 D_8007341F;
```
Comment (bismurphy) [inline src/weapon/w_034.c:10]: These should be PLAYER.hitboxWidth and height

File: src/weapon/w_012.c [hunk 2]
```diff
@@ -1,17 +1,447 @@
...
+        prim->priority = PLAYER.zPriority + 2;
 
-INCLUDE_ASM("weapon/nonmatchings/w_012", func_ptr_80170004);
+        D_58000_8017BB28++;
+        g_api.PlaySfx(SFX_WEAPON_SWISH_B);
+        g_Player.D_80072F00[9] = 4;
+        self->step++;
+    } else {
+        if (maskedParams == 0x100) {
+            DecelerateX(0x3000);
```
Comment (sozud) [inline src/weapon/w_012.c:268]: FIX? Same for the other

File: src/weapon/w_034.c [hunk 3]
```diff
@@ -62,7 +144,120 @@ static void EntityWeaponAttack(Entity* self) {
...
+            self->ext.sword.unk7C = 0x2000;
+        } else {
+            self->posX.i.hi = PLAYER.posX.i.hi + PLAYER.hitboxOffX;
+            self->ext.sword.unk7C = -0x2000;
+        }
+
+        self->velocityX += PLAYER.velocityX;
+        self->ext.sword.unk84 = 0x200;
+        self->velocityY = FIX(-0.25);
+        self->ext.sword.unk86 = 0x400;
```
Comment (sozud) [inline src/weapon/w_034.c:193]: Later gets added to velocity so could be fix

File: src/weapon/w_034.c [hunk 4]
```diff
@@ -62,7 +144,120 @@ static void EntityWeaponAttack(Entity* self) {
...
+        self->ext.sword.unk7E = 0x1C;
+        offsetY = PLAYER.posY.i.hi + PLAYER.hitboxOffY;
+        self->posY.i.hi = offsetY - 8;
+        if (PLAYER.step != 2) {
+            self->posY.i.hi = offsetY - 0x10;
+        }
+
+        if (self->facingLeft) {
+            self->posX.i.hi = PLAYER.posX.i.hi - PLAYER.hitboxOffX;
+            self->ext.sword.unk7C = 0x2000;
```
Comment (sozud) [inline src/weapon/w_034.c:184]: FIX

**Review by sozud — APPROVED** [review]

**Pull request merged**